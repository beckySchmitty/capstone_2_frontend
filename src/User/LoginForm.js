import React, { useState } from "react";
import { useHistory } from "react-router-dom";

import AlertMsg from "../helpers/AlertMsg"

// Login Form
// manages update to state on change
// On Submission - calls login function & redirects to /search

function LoginForm({ login }) {
  const history = useHistory();
  const [formData, setFormData] = useState({
    username: "",
    password: "",
  });

  const [errors, setErrors] = useState([]);


// HANDLE SUBMIT - redirect or error
  async function handleSubmit(e) {
    e.preventDefault();
    let resp = await login(formData);

    if (resp.success) {
      history.push("/search");
    } else {
      setErrors(resp.errors);
    }
  }

//  Form field updates on change
  function handleChange(e) {
    const { name, value } = e.target;
    setFormData(data => ({ ...data, [name]: value }));
  }

  return (
        <div>
          <h3>LOGIN</h3>

          <div>
            <div>
              <form onSubmit={handleSubmit}>
                <div>
                  <label>Username</label>
                  <input
                      name="username"
                      value={formData.username}
                      onChange={handleChange}
                      autoComplete="username"
                      required
                  />
                </div>
                <div>
                  <label>Password</label>
                  <input
                    name="password"
                      type="password"
                      value={formData.password}
                      onChange={handleChange}
                      autoComplete="current-password"
                      required
                    />
                    </div>

        {/* SHOW ERRORS IF ANY */}
                {errors.length
                    ? <AlertMsg type="danger" messages={errors} />
                    : null}

                <button onSubmit={handleSubmit}>Submit</button>
              </form>

            </div>
          </div>
        </div>
  );
}

export default LoginForm;
