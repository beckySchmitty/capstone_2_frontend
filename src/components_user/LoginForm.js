import React, { useState } from "react";
import { useHistory } from "react-router-dom";
import { Button, Form, FormGroup, Label, Input, Container, Col } from 'reactstrap';
import "../styles/Form.css"
import AlertMsg from "../helpers/AlertMsg"

// Login Form
// manages update to state on change
// On Submission - calls login function & redirects to /search

function LoginForm({ login }) {
  const history = useHistory();
  const [formData, setFormData] = useState({
    username: "",
    password: "",
  });

  const [errors, setErrors] = useState([]);


// HANDLE SUBMIT - redirect or error
  async function handleSubmit(e) {
    e.preventDefault();
    let resp = await login(formData);

    if (resp.success) {
      history.push("/search");
    } else {
      setErrors(resp.errors[0].data.error.message);
    }
  }

//  Form field updates on change
  function handleChange(e) {
    const { name, value } = e.target;
    setFormData(data => ({ ...data, [name]: value }));
  }

  return (
    <Container className="Form" >
    <h3>LOGIN</h3>
      <Form onSubmit={handleSubmit} className="Form-form">
      <Col>
      <FormGroup>
          <Label>Username</Label>
          <Input
              name="username"
              value={formData.username}
              onChange={handleChange}
              autoComplete="username"
              required
          />
      </FormGroup>
      </Col>
      <Col>
      <FormGroup>
          <Label>Password</Label>
          <Input
            name="password"
              type="password"
              value={formData.password}
              onChange={handleChange}
              required
            />
        </FormGroup>
        </Col>
        {errors.length
            ? <AlertMsg type="danger" error={errors} />
            : null}

        <Button onSubmit={handleSubmit}>Submit</Button>
      </Form>
    </Container>
  );
}

export default LoginForm;
