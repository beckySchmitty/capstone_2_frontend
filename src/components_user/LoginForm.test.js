import React from "react";
import { render } from "@testing-library/react";
import LoginForm from "./LoginForm";

// smoke test
it("renders without crashing", function() {
    render(<LoginForm />);
  });
  

// snapshot test
it("matches snapshot", function() {
  const {asFragment} = render(<LoginForm />);
  expect(asFragment()).toMatchSnapshot();
});

