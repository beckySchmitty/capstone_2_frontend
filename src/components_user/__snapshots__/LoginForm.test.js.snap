// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Form container"
  >
    <h3>
      LOGIN
    </h3>
    <form
      class="Form-form"
    >
      <div
        class="col"
      >
        <div
          class="form-group"
        >
          <label
            class=""
          >
            Username
          </label>
          <input
            autocomplete="username"
            class="form-control"
            name="username"
            required=""
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="col"
      >
        <div
          class="form-group"
        >
          <label
            class=""
          >
            Password
          </label>
          <input
            class="form-control"
            name="password"
            required=""
            type="password"
            value=""
          />
        </div>
      </div>
      <button
        class="btn btn-secondary"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
